11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/2.LOCO_data_split.py

```diff
@@ -0,0 +1,215 @@
+
+cell_type = args.cell_type
+
+
+# In[5]:
+
+
+# set path to the import data
+data_path = pathlib.Path(
+    f"../../../data/{cell_type}_preprocessed_sc_norm_aggregated_nomic.parquet"
+)
+# results path
+results_path = pathlib.Path("../results/").resolve()
+results_path.mkdir(exist_ok=True)
+
+# dataframe with only the labeled data we want (exclude certain phenotypic classes)
+data_df = pd.read_parquet(data_path)
+data_df.head()
+
+
+# In[6]:
+
+
+# get all of the features
+features = data_df.columns.to_list()
+
+nuclei_features = []
+pm_features = []
+gsdmd_features = []
+mito_features = []
+er_features = []
+other_features = []
+metadata_features = []
+correlation_features = []
+secretome_features = []
+
+# Separate the features into distinct types
+# code loop modified from:
+# https://github.com/WayScience/nuclear_speckles_analysis/blob/main/1.regression_modelling/1.train_models.ipynb
+# Thank you Jenna Tomkinson for the code!
+for feature in features:
+    # check if the feature contains "Metadata"
+    if "Metadata" in feature:
+        metadata_features.append(feature)
+    elif "[NSU]" in feature:
+        secretome_features.append(feature)
+    else:
+        parts = feature.split("_")
+
+        if "Correlation" in parts:  # Check if it's a correlation feature
+            correlation_features.append(feature)
+
+        else:  # Non-correlation features
+            if "CorrDNA" in parts:
+                nuclei_features.append(feature)
+            elif "CorrPM" in parts:
+                pm_features.append(feature)
+            elif "CorrGasdermin" in parts:
+                gsdmd_features.append(feature)
+            elif "CorrMito" in parts:
+                mito_features.append(feature)
+            elif "CorrER" in parts:
+                er_features.append(feature)
+            else:
+                other_features.append(feature)
+
+
+# In[7]:
+
+
+print(f"nuclei_features: {len(nuclei_features)}")
+print(f"pm_features: {len(pm_features)}")
+print(f"gsdmd_features: {len(gsdmd_features)}")
+print(f"mito_features: {len(mito_features)}")
+print(f"er_features: {len(er_features)}")
+print(f"other_features: {len(other_features)}")
+print(f"correlation_features: {len(correlation_features)}")
+print(f"metadata_features: {len(metadata_features)}")
+print(f"secretome_features: {len(secretome_features)}")
+print(f"total features: {len(features)}")
+# check if all featues are accounted for
+assert len(features) == len(nuclei_features) + len(pm_features) + len(
+    gsdmd_features
+) + len(mito_features) + len(er_features) + len(other_features) + len(
+    correlation_features
+) + len(
+    metadata_features
+) + len(
+    secretome_features
+)
+
+
+# In[8]:
+
+
+# create the combinations of features
+# out of 5 channels, we can have 0, 1, 2, 3, 4, or 5 channels
+# even with 0 channels, we still have the non channel features
+# these are areshape features
+# set the feature combination lists
+dict_of_feature_combinations = {
```

Comment:
why are there so many combinations? Shouldn't there just be five combinations? Maybe this is why the analysis is taking so long?